x/storage/keeper/cross_app_permission.go

```diff
@@ -0,0 +1,222 @@
+package keeper
+
+import (
+	"encoding/json"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	permtypes "github.com/bnb-chain/greenfield/x/permission/types"
+	"github.com/bnb-chain/greenfield/x/storage/types"
+)
+
+var _ sdk.CrossChainApplication = &PermissionApp{}
+
+type PermissionApp struct {
+	permissionKeeper types.PermissionKeeper
+}
+
+func NewPermissionApp(keeper types.PermissionKeeper) *PermissionApp {
+	return &PermissionApp{
+		permissionKeeper: keeper,
+	}
+}
+
+func (app *PermissionApp) ExecuteAckPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	pack, err := types.DeserializeCrossChainPackage(payload, types.PermissionChannelId, sdk.AckCrossChainPackageType)
+	if err != nil {
+		panic("deserialize Policy cross chain package error")
+	}
+
+	var operationType uint8
+	var result sdk.ExecuteResult
+	switch p := pack.(type) {
+	case *types.CreatePolicyAckPackage:
```

Comment:
Why we have Ack Packages?  Please explain in what case we may receive ACK Package???

Reply:
removed